File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        sum1=0
        M=len(mat)
        for i in range(len(mat)):
            for j in range(len(mat[0])):
                if(i==j):
                    sum1+=mat[i][j]
                elif i+j==M-1:
                    sum1=sum1+mat[i][j]
        return sum1
            
        

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int n= mat.length;
        int sum1=0;
        for(int i =0;i<mat.length;i++)
        {
            for(int j=0;j<mat[0].length;j++)
            {
                if(i==j)
                {
                    sum1+=mat[i][j];
                }
                if(i!=j && (i+j)==n-1)
                {
                    sum1+=mat[i][j];
                    
                }
            }
        }
        return sum1;
        
    }
}